tests/queries/0_stateless/01595_countMatches.sql

```diff
@@ -0,0 +1,21 @@
+select 'case sensitive';
+select countMatches('foobarfoo', 'foo');
+select countMatches('foobarfoo', 'foo.*');
+select countMatches('oooo', 'oo');
+select countMatches(concat(toString(number), 'foofoo'), 'foo') from numbers(2);
+select countMatches('foobarbazfoobarbaz', 'foo(bar)(?:baz|)');
+select countMatches('foo.com bar.com baz.com bam.com', '([^. ]+)\.([^. ]+)');
+select countMatches('foo.com@foo.com bar.com@foo.com baz.com@foo.com bam.com@foo.com', '([^. ]+)\.([^. ]+)@([^. ]+)\.([^. ]+)');
```

Comment:
We probably should add it to `constant_column_search` perf test and compare against `length(multiMatchAllIndices)`, this one should be faster. If not, maybe we don't even need a separate function.

Reply:
>We probably should add it to constant_column_search perf test and compare against length(multiMatchAllIndices)

Good idea.

>this one should be faster. If not, maybe we don't even need a separate function.

Actually hyperscan works faster

```sql
SELECT countIf(1, length(multiMatchAllIndices(URL, ['(yandex.*|ya)\\.(com|ru)'])) > 0) AS cnt
FROM hits

┌─────cnt─┐
│ 2115167 │
└─────────┘

1 rows in set. Elapsed: 0.265 sec. Processed 8.87 million rows, 762.68 MB (33.47 million rows/s., 2.88 GB/s.) 
```

```sql
SELECT countIf(1, countMatches(URL, '(yandex.*|ya)\\.(com|ru)') > 0) AS cnt
FROM hits

┌─────cnt─┐
│ 2115167 │
└─────────┘

1 rows in set. Elapsed: 0.631 sec. Processed 8.87 million rows, 762.68 MB (14.06 million rows/s., 1.21 GB/s.) 
```

So maybe it worth just add something like `multiMatchCount`, to avoid creating array and return just a number of matches.

Reply:
Hyperscan is unsafe for user-provided regexps, thus the setting `allow_hyperscan` exists.
It is faster than re2 though. But re2 is battle tested in Yandex.Metrica and it never caused DoS.

We should keep the `countMatches` function.

Reply:
>We should keep the countMatches function.

Ok